exp/controllers/machinepool_controller_phases.go

```diff
@@ -65,6 +65,16 @@ func (r *MachinePoolReconciler) reconcilePhase(mp *expv1.MachinePool) {
 		mp.Status.SetTypedPhase(expv1.MachinePoolPhaseRunning)
 	}
 
+	// Set the phase to "scalingUp" if the infrastructure is scaling up.
+	if mp.Status.InfrastructureReady && *mp.Spec.Replicas > mp.Status.Replicas {
+		mp.Status.SetTypedPhase(expv1.MachinePoolPhaseScalingUp)
+	}
+
+	// Set the phase to "scalingDown" if the infrastructure is scaling down.
+	if mp.Status.InfrastructureReady && *mp.Spec.Replicas < mp.Status.Replicas {
```

Comment:
I'm not entirely sure if Status.Replicas is the best way to describe "current" replicas (Spec.Replicas being desired), the Machine Deployment phases do something different with unavailable replicas:

```
	// This is the same as unavailableReplicas, but we have to recalculate because unavailableReplicas
	// would have been reset to zero above if it was negative
	if totalReplicas-availableReplicas < 0 {
		status.Phase = string(clusterv1.MachineDeploymentPhaseScalingDown)
```
https://github.com/kubernetes-sigs/cluster-api/blob/master/controllers/machinedeployment_sync.go#L382-L383